integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala

```diff
@@ -510,6 +509,7 @@ class CarbonFileMetastore extends CarbonMetaStore {
         SegmentPropertiesAndSchemaHolder.getInstance().invalidate(absoluteTableIdentifier)
       }
     }
+    removeTableFromMetadata(dbName, tableName)
```

Comment:
Moving outside can be a problem as according to above logic inside else block there is again an if check based on which this method is getting called. With current check that if check will not be used for decision making